shell/platform/embedder/embedder_semantics_update.h

```diff
@@ -58,13 +58,23 @@ class EmbedderSemanticsUpdate2 {
   std::vector<FlutterSemanticsCustomAction2> actions_;
   std::vector<FlutterSemanticsCustomAction2*> action_pointers_;
 
+  std::vector<std::unique_ptr<std::vector<FlutterStringAttribute*>>>
```

Comment:
I am no longer super familiar with this bit of the codebase. But can you conform that the underlying nodes array is never modified after the vector of pointers is constructed? This will invalidate the old pointers.

Reply:
> But can you conform that the underlying nodes array is never modified after the vector of pointers is constructed?

Yup, confirmed. Pointers are captured only after the vector that owns the memory is fully constructed. I mirrored the approach taken for the layer tree: https://github.com/flutter/engine/blob/f1f912c0c7c78a42248862396b06f796e457310f/shell/platform/embedder/embedder_layers.h#L40-L45